.github/workflows/tests.yaml

```diff
@@ -0,0 +1,139 @@
+name: Unit and E2E Tests
+on: [pull_request]
+jobs:
+
+#### Unit tests ####
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go 1.22
+        uses: actions/setup-go@v5
+        with:
+          go-version: 1.22
+      - name: Install dependencies
+        run: |
+          go mod tidy
+          go install golang.org/x/tools/cmd/cover@v0.1.0
+      - name: Test CFN lambda
+        working-directory: ./cfn-lambda/handler
+        run: go test -v -race -covermode=atomic -coverprofile=coverage.out
+      - name: Test EventBridge lambda
+        working-directory: ./eventbridge-lambda/handler
+        run: go test -v -race -covermode=atomic -coverprofile=coverage.out
+      - name: Test common functions
+        working-directory: ./common
+        run: go test -v -race -covermode=atomic -coverprofile=coverage.out
+
+#### E2E Tests ####
+  e2e-tests:
+    needs: unit-tests  # Run e2e test only if unit tests passed
+    runs-on: ubuntu-latest
+    env:
+      AWS_REGION: 'us-east-1'
+      TEST_VERSION: 0.0.7357
```

Comment:
If I understand correctly this is a arbitrary number to tag the test version right? would it accept strings? I think we can add `test` suffix or prefix here